### PR TITLE
[CI-APP] Update CI Pipeline documentation

### PR DESCRIPTION
**Pull request by TovRudyy**
### What does this PR do? What is the motivation?

Add a couple of minor clarifications about CI Pipeline Visibility and two extra troubleshooting sections. 

Our motivation is to avoid confusion among our users, as we have received some complaints about behaviours totally normal that were reported as problems.

### Merge instructions

- [x] Please merge after reviewing

### CODE REVIEW

File: content/en/continuous_integration/guides/pipeline_data_model.md [hunk 1]
```diff
@@ -33,6 +33,8 @@ When a pipeline, stage, job, or step finishes, execution data is sent to Datadog
 
 {{< img src="ci/ci-pipeline-execution.png" alt="Example of a pipeline execution trace" style="width:100%;">}}
 
+Stages, jobs, and steps are expected to have the same pipeline name as their parent pipeline. In case this is not true, there might be missing information on the user interface. For example, missing jobs in the job summary tables.
```
Comment (kassenq) [inline content/en/continuous_integration/guides/pipeline_data_model.md:36]: exact same pipeline name as their parent pipeline

In the case of a mismatch, some pipelines may be missing stage, job, and step information.

File: content/en/continuous_integration/search/_index.md [hunk 2]
```diff
@@ -40,7 +40,7 @@ You can access high-level accumulation and trends, including:
 
 To see your pipelines, navigate to [**CI** > **Pipelines**][101].
 
-The [Pipelines page][101] shows aggregate stats for the default branch of each pipeline over the selected time frame, as well as the status of the latest pipeline execution. Use this page to see all your pipelines and get a quick view of their health. The Pipelines page shows metrics for the _default_ branch, usually named something like `main` or `prod`.
+The [Pipelines page][101] shows aggregate stats of your pipelines as well as the status of the latest pipeline execution. For pipelines with Git information, it only shows stats for the default branch of each pipeline over the selected time frame. The _default_ branch is usually named something like `main` or `prod`. Use this page to see all your pipelines and get a quick view of their health.
```
Comment (kassenq) [inline content/en/continuous_integration/search/_index.md:43]: shows aggregate stats for the default branch of each pipeline over the selected time frame, as well as the status of the latest pipeline execution. Use this page to see all your pipelines and get a quick view of their health. Only pipelines with Git information associated to the _default_ branch (usually named something like `main` or `prod`), as well as pipelines without any Git information, are displayed on this page.

File: content/en/continuous_integration/troubleshooting.md [hunk 3]
```diff
@@ -175,6 +175,14 @@ If you have admin access, you can update it from the [Repository Settings Page][
 
 A "Pipeline not found" message is shown when you click on incomplete data coming from an in-progress pipeline. Data is received progressively for stages, jobs, or custom commands. Wait until the pipeline has finished and try again.
 
+### Missing pipelines on the pipeline list
```
Comment (kassenq) [inline content/en/continuous_integration/troubleshooting.md:178]: ### Missing pipelines on the Pipelines page

File: content/en/continuous_integration/troubleshooting.md [hunk 4]
```diff
@@ -175,6 +175,14 @@ If you have admin access, you can update it from the [Repository Settings Page][
...
+
+The pipeline list only displays pipelines with no Git information, or pipelines with Git information which belong to the default branch of the Git repository.
+
+### Missing jobs in the job summary table
```
Comment (kassenq) [inline content/en/continuous_integration/troubleshooting.md:182]: ### Missing stages or jobs in summary tables

File: content/en/continuous_integration/troubleshooting.md [hunk 5]
```diff
@@ -175,6 +175,14 @@ If you have admin access, you can update it from the [Repository Settings Page][
...
+
+Missing jobs on the _Pipeline Details_ page might be due to a wrong configuration of the jobs. Make sure that the pipeline name stored in the job executions matches the name of their parent pipeline.
```
Comment (kassenq) [inline content/en/continuous_integration/troubleshooting.md:184]: If they don't match, should we direct the customer to the attribute in the API documentation so they know to update it if they're sending custom pipelines?

**Review by kassenq — CHANGES_REQUESTED** [review]

File: content/en/continuous_integration/troubleshooting.md (hunk 5 above)
Comment (TovRudyy) [reply content/en/continuous_integration/troubleshooting.md]: Sounds good. I added a link to it.

File: content/en/continuous_integration/search/_index.md [hunk 6]
```diff
@@ -40,7 +40,7 @@ You can access high-level accumulation and trends, including:
 
 To see your pipelines, navigate to [**CI** > **Pipelines**][101].
 
-The [Pipelines page][101] shows aggregate stats for the default branch of each pipeline over the selected time frame, as well as the status of the latest pipeline execution. Use this page to see all your pipelines and get a quick view of their health. The Pipelines page shows metrics for the _default_ branch, usually named something like `main` or `prod`.
+The [Pipelines page][101] shows aggregate stats for the default branch of each pipeline over the selected time frame, as well as the status of the latest pipeline execution. Use this page to see all your pipelines and get a quick view of their health. Only pipelines with Git information associated to the default branch (usually named something like `main` or `prod`), as well as pipelines without any Git information, are displayed on this page.
```
Comment (aliciascott) [inline content/en/continuous_integration/search/_index.md:43]:
```suggestion
The [Pipelines page][101] shows aggregate stats for the default branch of each pipeline over the selected time frame, as well as the status of the latest pipeline execution. Use this page to see all your pipelines and get a quick view of their health. Only pipelines with Git information associated to the default branch (usually named `main` or `prod`), as well as pipelines without any Git information, are displayed on this page.
```

File: content/en/continuous_integration/troubleshooting.md [hunk 7]
```diff
@@ -175,6 +175,14 @@ If you have admin access, you can update it from the [Repository Settings Page][
 
 A "Pipeline not found" message is shown when you click on incomplete data coming from an in-progress pipeline. Data is received progressively for stages, jobs, or custom commands. Wait until the pipeline has finished and try again.
 
+### Missing pipelines on the Pipelines page
+
+The pipeline page only displays pipelines with no Git information, or pipelines with Git information which belong to the default branch of the Git repository.
+
+### Missing stages or jobs in summary tables
+
+Missing stages or jobs in the _Pipeline Details_ page might be due to a wrong configuration. Make sure that the pipeline name stored in the stage or job executions matches the exact same name of their parent pipeline. If you are using custom pipelines, refer to the [public API endpoint specification][15].
```
Comment (aliciascott) [inline content/en/continuous_integration/troubleshooting.md:184]:
```suggestion
Missing stages or jobs in the _Pipeline Details_ page might be due to a wrong configuration. Make sure that the pipeline name stored in the stage or job executions matches the **same** name of their parent pipeline. If you are using custom pipelines, refer to the [public API endpoint specification][15].
```

**Review by aliciascott — CHANGES_REQUESTED** [review]
Just a couple of minor suggestions for wording changes, otherwise looks good. Please request re-review when changes are submitted, thanks!

**Review by aliciascott — APPROVED** [review]
looks good !

**Pull request merged**